torchqdynamics/sesolve/sesolve.py

```diff
@@ -0,0 +1,66 @@
+from typing import List, Literal, Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from ..odeint import odeint
+from ..solver import SolverOption
+from ..types import OperatorLike, TensorLike, TimeDependentOperatorLike, to_tensor
+
+
+def sesolve(
+    H: TimeDependentOperatorLike,
+    psi: OperatorLike,
+    t_save: TensorLike,
+    *,
+    exp_ops: Optional[List[OperatorLike]] = None,
+    save_states: bool = True,
+    gradient_alg: Literal[None, 'autograd', 'adjoint'] = None,
```

Comment:
I prefer `Optional[Literal['autograd', 'adjoint']]` here, to make clear that the sensitivity algorithm is optional.